### PR TITLE
[KYUUBI 3269] [DOCS] Doc for JDBC authentication provider

### PR DESCRIPTION
**Pull request by bowenliang123**
### _Why are the changes needed?_

fix https://github.com/apache/incubator-kyuubi/issues/3269

Add docs for JDBC auth provider from https://github.com/apache/incubator-kyuubi/issues/3222 shipping within 1.6.0, add jdbc.rst in /docs/security path
simplify the config docs for kyuubi.authentication.jdbc.query in /docs/deployment/settings.md by moving examples to the docs of jdbc auth provider.

### _How was this patch tested?_
- [ ] Add some test cases that check the changes thoroughly including negative and positive cases if possible

- [ ] Add screenshots for manual tests if appropriate

- [x] [Run test](https://kyuubi.apache.org/docs/latest/develop_tools/testing.html#running-tests) locally before make a pull request

### CODE REVIEW

File: kyuubi-common/src/main/scala/org/apache/kyuubi/config/KyuubiConf.scala [hunk 1]
```diff
@@ -707,14 +707,7 @@ object KyuubiConf {
...
         "Authentication passes if at least one row fetched in the result set." +
-        "Available placeholders are: <ul>" +
-        "<li>`${username}`</li>" +
-        "<li>`${password}`</li></ul>" +
-        "eg.: query sql `SELECT 1 FROM auth_table WHERE user=${username} AND " +
-        "passwd=MD5(CONCAT(salt,${password}));` " +
-        "will be prepared as: `SELECT 1 FROM auth_table " +
-        "WHERE user=? AND passwd=MD5(CONCAT(salt,?));`" +
-        " with value replacement of `username` and `password` in string type.")
+        "Available placeholders are `${username}` and `${password}`.")
```
Comment (pan3793) [inline kyuubi-common/src/main/scala/org/apache/kyuubi/config/KyuubiConf.scala:710]: The docs should be regenerated

File: docs/security/jdbc.md [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+To enable the jdbc authentication method, we need to
+
+- put the JDBC driver jar package to ``$KYUUBI_HOME/jars`` directory to make it visible for
+  the classpath of the kyuubi server.
+- Configure the following properties to ``$KYUUBI_HOME/conf/kyuubi-defaults.conf``
+  on each node where kyuubi server is installed.
+
+
+## Configure the authentication properties
+Configure the following properties to `$KYUUBI_HOME/conf/kyuubi-defaults.conf` on each node where kyuubi server is installed.
```
Comment (pan3793) [inline docs/security/jdbc.md:41]: nit: empty line

File: docs/security/jdbc.md [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+- `${password}`
+
+For example, `SELECT 1 FROM auth_db.auth_table WHERE user=${username} AND 
+passwd=MD5(CONCAT(salt,${password}))` will be prepared as: `SELECT 1 FROM auth_db.auth_table WHERE user=? AND passwd=MD5(CONCAT(salt,?))` with value replacement of `username` and `password` in string type.
+
+## Enable JDBC Authentication 
+
+To enable the jdbc authentication method, we need to
+
+- put the JDBC driver jar package to ``$KYUUBI_HOME/jars`` directory to make it visible for
```
Comment (pan3793) [inline docs/security/jdbc.md:34]: UPPER CASE?

Comment (bowenliang123) [reply docs/security/jdbc.md]: sure

File: docs/security/jdbc.md (hunk 2 above)
Comment (bowenliang123) [reply docs/security/jdbc.md:40]: ok

File: kyuubi-common/src/main/scala/org/apache/kyuubi/config/KyuubiConf.scala (hunk 1 above)
Comment (bowenliang123) [reply kyuubi-common/src/main/scala/org/apache/kyuubi/config/KyuubiConf.scala:710]: my mistake. regenerated.

File: docs/security/jdbc.md (hunk 3 above)
Comment (bowenliang123) [reply docs/security/jdbc.md]: alright. lowered.

**Review by pan3793 — APPROVED** [review]

File: docs/security/jdbc.md [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+ -
+ - Unless required by applicable law or agreed to in writing, software
+ - distributed under the License is distributed on an "AS IS" BASIS,
+ - WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ - See the License for the specific language governing permissions and
+ - limitations under the License.
+ -->
+
+
+# Configure Kyuubi to use JDBC Authentication
```
Comment (yaooqinn) [inline docs/security/jdbc.md:19]: Use

File: docs/security/jdbc.md [hunk 5]
```diff
@@ -0,0 +1,49 @@
...
+ - distributed under the License is distributed on an "AS IS" BASIS,
+ - WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ - See the License for the specific language governing permissions and
+ - limitations under the License.
+ -->
+
+
+# Configure Kyuubi to Use JDBC Authentication
+
+Kyuubi supports authentication via JDBC query. A query is prepared with username/password value and sent to database configured in JDBC URL. The authentication passes if result set returns at least one row.
```
Comment (yaooqinn) [inline docs/security/jdbc.md:21]:
```suggestion
Kyuubi supports authentication via JDBC query. A query is prepared with username/password value and sent to the database configured in JDBC URL. The authentication passes if the result set is not empty.
```

File: docs/security/jdbc.md [hunk 6]
```diff
@@ -0,0 +1,49 @@
...
+
+The query sql must start with `SELECT`. Placeholders are supported for sql template:
```
Comment (yaooqinn) [inline docs/security/jdbc.md:23]: The SQL statement must start with the `SELECT` clause. Placeholders are supported and listed below for substitution

**Review by yaooqinn — APPROVED** [review]

**Comment by yaooqinn** [comment]
thanks @bowenliang123 for the contribution and @pan3793 for the review

merged to master

**Pull request closed without merging**